pkg/apis/nmstate/v1/nodenetworkstate_types_test.go

```diff
@@ -0,0 +1,105 @@
+package v1
+
+import (
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	yaml "sigs.k8s.io/yaml"
+)
+
+var _ = Describe("NodeNetworkState", func() {
+	var (
+		desiredState = State(`
+interfaces:
+  - name: eth1
+    type: ethernet
+    state: up`)
+
+		currentState = State(`
+interfaces:
+  - name: eth1
+    type: ethernet
+    state: down`)
+
+		nnsManifest = `
+apiVersion: nmstate.io/v1
+kind: NodeNetworkState
+metadata:
+  name: node01
+  creationTimestamp: "1970-01-01T00:00:00Z"
+spec:
+  managed: true
+  nodeName: node01
+  desiredState:
+    interfaces:
+      - name: eth1
+        type: ethernet
+        state: up
+status:
+  currentState:
+    interfaces:
+      - name: eth1
+        type: ethernet
+        state: down
+
+`
+		nnsStruct = NodeNetworkState{
+			TypeMeta: metav1.TypeMeta{
+				APIVersion: "nmstate.io/v1",
+				Kind:       "NodeNetworkState",
+			},
+			ObjectMeta: metav1.ObjectMeta{
+				Name:              "node01",
+				CreationTimestamp: metav1.Unix(0, 0),
+			},
+			Spec: NodeNetworkStateSpec{
+				Managed:      true,
+				NodeName:     "node01",
+				DesiredState: desiredState,
+			},
+			Status: NodeNetworkStateStatus{
+				CurrentState: currentState,
+			},
+		}
+	)
+
+	Context("when reading NodeNetworkState from a raw manifest", func() {
+
+		var nodeNetworkStateStruct NodeNetworkState
+
+		BeforeEach(func() {
+			err := yaml.Unmarshal([]byte(nnsManifest), &nodeNetworkStateStruct)
```

Comment:
sorry, maybe I'm missing something. Why do we need to Unmarshal (and Marshal in the other test) the manifest before every test, wouldn't be defining it in the scope of `Describe("NodeNetworkState"` enough?

Reply:
We have to test to different thinks marshal and unmarshal so we have two context for each, and it context to their thing at BeaforeEach.